Assets/Scripts/Enemy/DroneFlock.cs

```diff
@@ -0,0 +1,204 @@
+
+    [SerializeField] private float alignFactor = 1f;
+    [SerializeField] private float cohesionFactor = 1f;
+    [SerializeField] private float seekFactor = 4f;
+    private List<DroneEnemy> queuedDrones = new List<DroneEnemy>();
+
+    private void AddDrone(DroneEnemy drone)
+    {
+      queuedDrones.Add(drone);
+    }
+
+    private void Update()
+    {
+      UpdateFlock();
+    }
+
+    private void UpdateFlock()
+    {
+      if (queuedDrones.Count > 0)
+      {
+        foreach (var drone in queuedDrones)
+        {
+          flockingDrones.Add(drone);
+        }
+
+        queuedDrones.Clear();
+      }
+
+      for (int i = 0; i < flockingDrones.Count; i++)
+      {
+        Vector3 sep = Separate(i); // Separation	
+        Vector3 ali = Align(i); // Alignment	
+        Vector3 coh = Cohesion(i); // Cohesion	
+        Vector3 seek = Seek(i, playerTransform.position);
+        // Arbitrarily weight these forces	
+        sep *= separationFactor;
+        ali *= alignFactor;
+        coh *= cohesionFactor;
+        seek *= seekFactor;
+        // Add the force vectors to acceleration	
+        flockingDrones[i].ApplyForce(sep + ali + coh + seek);
+      }
+    }
+
+    private void LimitVector(ref Vector3 vector, float maxValue)
+    {
+      if (vector.magnitude > maxValue)
+      {
+        vector.Normalize();
+        vector *= maxValue;
+      }
+    }
+
+    private Vector3 Seek(int index, Vector3 target)
+    {
+      Vector3 dronePosition = flockingDrones[index].rb.position;
+      dronePosition.y = 0;
+      target.y = 0;
+      
+      Vector3
+        desired = target - dronePosition; // A vector pointing from the position to the target	
+      // Scale to maximum speed	
+      desired.Normalize();
+      desired *= aiMovementStats.MaxVelocity.Value;
+      // Above two lines of code below could be condensed with new PVector setMag() method	
+      // Not using this method until Processing.js catches up	
+      // desired.setMag(maxspeed);	
+      // Steering = Desired minus Velocity	
+      Vector3 steer = desired - flockingDrones[index].rb.velocity;
+      // Limit to maximum steering force	
+      LimitVector(ref steer, aiMovementStats.AccelerationForce.Value);
+      return steer;
+    }
+
+    private Vector3 Separate(int index)
+    {
+      Vector3 position = flockingDrones[index].rb.position;
+      Vector3 steer = new Vector3(0, 0, 0);
+      int inRangeCount = 0;
+      for (int i = 0; i < flockingDrones.Count; i++)
+      {
+        if (i == index) continue;
+        // For every boid in the system, check if it's too close	
+        float d = Vector3.Distance(position, flockingDrones[i].rb.position);
+        // If the distance is greater than 0 and less than an arbitrary amount (0 when you are yourself)	
+        if (d < desiredSeparation)
+        {
+          // Calculate vector pointing away from neighbor	
+          Vector3 diff = position - flockingDrones[i].rb.position;
+          diff.Normalize();
```

Comment:
You are using Normalize a lot in your code, would be good to already start thinking on how to optimise this